src/flag_gems/ops/sort.py

```diff
@@ -66,7 +68,9 @@ def floating_to_uint(x, descending: tl.constexpr = False):
 
     sign_bit_mask: tl.constexpr = one_zeros(num_bits)
     # mind the dtype, right_shift for signed is arithmetic right shift
-    mask = sign_bit_mask | (sx >> (num_bits - 1)).to(udtype, bitcast=True)
+    # Fix for triton 3.1 or else `sx >> rshift_bits` is promoted to int32
+    rshift_bits = tl.full((), value=num_bits - 1, dtype=sdtype)
+    mask = sign_bit_mask | (sx >> rshift_bits).to(udtype, bitcast=True)
```

Comment:
what about just right shifting the sign_bit_mask >> rshift_bits ?

Reply:
`rshift_bits` is a scalar in int32, thus `sx >> rshift_bits` gets promoted to int32. When sx is of tl.int16, it gets an error when trying to bitcast it to tl.uint16.